### PR TITLE
[IMP] web: drill down to records in chart on click

### PR DESCRIPTION
**Pull request by dhp-odoo**
PURPOSE:

Drilling down the records by providing the list of all the corresponding records
when we click on any bar in bar chart or any piece of pie in pie chart.

SPEC:

Passed domain parameter along with rest of the options like value, labels and
originIndex. Then used that specific domain for specific bar or pie in onClick event.
and by handling the click event, just open the list view of the records based on the 
specific domain get from the bar or piece of the pie. 

Task : 2198461

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: doc/reference/views.rst [hunk 1]
```diff
@@ -1072,6 +1072,8 @@ attributes:
 ``stacked``
   only used for ``bar`` charts. If present and set to ``True``, stacks bars
   within a group
+``disable_linking``
```
Comment (Arcasias) [inline doc/reference/views.rst]: Need to add this attribute to `graph_view.rng`

File: addons/web/static/tests/views/graph_tests.js [hunk 2]
```diff
@@ -979,6 +979,150 @@ QUnit.module('Views', {
...
+        assert.strictEqual(graph.renderer.state.mode, "bar", "should be in bar chart mode");
+        assert.checkDatasets(graph, ['domain'], {
+            domain: [[["bar", "=", true]], [["bar", "=", false]]],
+        });
+
+        let myChart = graph.renderer.chart;
+        let meta = myChart.getDatasetMeta(0);
+        let rectangle = myChart.canvas.getBoundingClientRect();
+        let point = meta.data[0].getCenterPoint();
+        let mouseMoveEvent = new MouseEvent('click', {
```
Comment (Arcasias) [inline addons/web/static/tests/views/graph_tests.js]: Better to use existing helpers: `testUtils.dom.triggerEvent` with `MouseEvent` typed events allows to provide an `eventAttrs` dict as third argument which can set `clientX` and `clientY` keys to the final triggered event object

Comment (Arcasias) [reply addons/web/static/tests/views/graph_tests.js]: And the event here is called `mouseMoveEvent` while it triggers a click...

File: addons/web/static/src/js/views/graph/graph_renderer.js [hunk 3]
```diff
@@ -987,6 +1011,26 @@ return AbstractRenderer.extend({
     // Handlers
     //--------------------------------------------------------------------------
 
+    /**
+     * @private
+     * @param {CustomEvent} ev
```
Comment (Arcasias) [inline addons/web/static/src/js/views/graph/graph_renderer.js]: `MouseEvent`

File: addons/web/static/tests/views/graph_tests.js [hunk 4]
```diff
@@ -979,6 +979,150 @@ QUnit.module('Views', {
...
+            View: GraphView,
+            model: "foo",
+            data: this.data,
+            arch: '<graph><field name="bar"/></graph>',
+            intercepts: {
+                do_action: function (ev) {
+                    assert.deepEqual(ev.data.action, {
+                        context: {},
+                        domain: [["bar", "=", true]],
+                        name: true, // ?
```
Comment (Arcasias) [inline addons/web/static/tests/views/graph_tests.js]: ?

File: doc/reference/views.rst (hunk 1 above)
Comment (aab-odoo) [reply doc/reference/views.rst]: bien vu...

**Comment by aab-odoo** [comment]
robodoo r+

**Comment by aab-odoo** [comment]
robodoo retry

**Pull request closed without merging**